remote_attestation/rust/Cargo.toml

```diff
@@ -7,45 +7,25 @@ license = "Apache-2.0"
 
 [features]
 default = ["ring-crypto"]
-std = ["anyhow/std", "prost/std"]
+std = ["anyhow/std"]
 ring-crypto = ["ring"]
-rust-crypto = [
-  "aes-gcm",
-  "getrandom/rdrand",
-  "hkdf",
-  "p256",
-  "rand_core",
-  "sha2",
-  "signature",
-  "x25519-dalek",
-]
+rust-crypto = ["aes-gcm", "hkdf", "p256", "sha2", "signature", "x25519-dalek"]
 
 [dependencies]
 aes-gcm = { version = "*", optional = true }
 anyhow = { version = "*", default-features = false }
 bytes = { version = "*", default-features = false }
-getrandom = { version = "*", optional = true }
 hkdf = { version = "*", optional = true }
-log = "*"
 p256 = { version = "*", default-features = false, optional = true, features = [
   "ecdsa"
 ] }
-prost = { version = "*", default-features = false, features = ["prost-derive"] }
-# Restrict the rand_core to a version that is compatible with x25519-dalek.
-rand_core = { version = "*", optional = true, default-features = false, features = [
```

Comment:
I thought the getrandom and rand_core dependencies are needed for the bare-metal binary, so I am surprised that the bare-metal binaries build.

Reply:
Soo this was... complicated. It turns out that our remote attestation crate only actually includes the rust-crypto parts if rust-crypto is enabled *and* ring-crypto is not:

https://github.com/project-oak/oak/blob/241a59e4dd5f1e405822daff190a77f4333893ad/remote_attestation/rust/src/crypto/mod.rs#L27-L28

And there is a compile time check that at least one is enabled:

https://github.com/project-oak/oak/blob/241a59e4dd5f1e405822daff190a77f4333893ad/remote_attestation/rust/src/crypto/mod.rs#L22-L23

What is *not* there is a similar check to ensure that they are not both enabled at the same time (which I now added).

In practice they were both getting enabled, and therefore the ring-crypto would be chosen. Still not entirely sure how that actually worked though.

Reply:
cc @andrisaar